src/borrower/nftfeed/nftfeed.sol

```diff
@@ -0,0 +1,126 @@
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.3;
+
+import "ds-note/note.sol";
+import "tinlake-auth/auth.sol";
+import "tinlake-math/math.sol";
+
+contract ShelfLike {
+    function shelf(uint loan) public view returns (address registry, uint tokenId);
+}
+
+contract PileLike {
+    function setRate(uint loan, uint rate) public;
+    function debt(uint loan) public returns (uint);
+}
+
+contract NFTFeed is DSNote, Auth, Math {
+    // nftID => nftValues
+    mapping (bytes32 => uint) public nftValues;
+
+    // nftID => risk
+    mapping (bytes32 => uint) public risk;
+
+    // risk => thresholdRatio
+    mapping (uint => uint) public thresholdRatio;
+    // risk => ceilingRatio
+    mapping (uint => uint) public ceilingRatio;
+    // risk => rate
+    mapping (uint => uint) public rate;
+
+    PileLike pile;
+    ShelfLike shelf;
+
+    /// defines default values for risk group 0
+    /// all values are denominated in RAY (10^27)
+    constructor (uint defaultThresholdRatio, uint defaultCeilingRatio, uint defaultRate) public {
+        thresholdRatio[0] = defaultThresholdRatio;
+        ceilingRatio[0] = defaultCeilingRatio;
+        rate[0] = defaultRate;
+    }
+
+
+    /// sets the dependency to another contract
+    function depend(bytes32 contractName, address addr) external auth {
+        if (contractName == "pile") {pile = PileLike(addr);}
+        else if (contractName == "shelf") { shelf = ShelfLike(addr); }
+        else revert();
+    }
+
+    function nftID(address registry, uint tokenId) public pure returns (bytes32) {
+        return keccak256(abi.encodePacked(registry, tokenId));
+    }
+
+    function nftID(uint loan) public view returns (bytes32) {
+        (address registry, uint tokenId) = shelf.shelf(loan);
+        return nftID(registry, tokenId);
+    }
+
+    /// Admin -- Updates
+    function setRiskGroup(uint risk_, uint thresholdRatio_, uint ceilingRatio_, uint rate_) public auth {
+        thresholdRatio[risk_] = thresholdRatio_;
+        ceilingRatio[risk_] = ceilingRatio_;
+        rate[risk_]= rate_;
+    }
+
+    function file(bytes32 what, uint risk_, uint value_) public auth {
+        if (what == "threshold") { thresholdRatio[risk_]  = value_; }
+        else if (what == "ceiling") { ceilingRatio[risk_] = value_; }
+        else if (what == "rate") { rate[risk_]= value_;}
+        else revert("unknown parameter");
+    }
```

Comment:
We could allow the admin to update individually ceiling, rate and threshold.

(If we allow it, we need three requires in the `update` method below  because not all values of a risk group are set)